packages/devtools/lib/src/memory/memory.dart

```diff
@@ -362,38 +650,291 @@ class MemoryScreen extends Screen with SetStateMixin {
       );
     }
 
-    table.onSelect.listen((InstanceSummary row) async {
-      ga.select(ga.memory, ga.inspectInstance);
+    table.onCellHover.listen(hoverInstanceAllocations);
+    table.onSelect.listen(select);
 
-      // User selected a new instance from the list of class instances so the
-      // instance view which would be the third child needs to be removed.
-      _removeInstanceView();
+    return table;
+  }
 
-      Instance instance;
-      try {
-        instance = await memoryController.getObject(row.objectRef);
-      } catch (e) {
-        instance = null; // Signal a problem
-      } finally {
-        tableContainer.add(_createInstanceView(
-          instance != null
-              ? row.objectRef
-              : 'Unable to fetch instance ${row.objectRef}',
-          row.className,
-        ));
-
-        tableContainer.element.scrollTo(<String, dynamic>{
-          'left': tableContainer.element.scrollWidth,
-          'top': 0,
-          'behavior': 'smooth',
-        });
+  void select(InstanceSummary row) async {
+    ga.select(ga.memory, ga.inspectInstance);
+
+    // User selected a new instance from the list of class instances so the
+    // instance view which would be the third child needs to be removed.
+    _removeInstanceView();
 
-        // Allow inspection of the memory object.
-        memoryDataView.showFields(instance != null ? instance.fields : []);
+    Instance instance;
+    try {
+      instance = await memoryController.getObject(row.objectRef);
+
+      final dynamic theObject = await memoryController.getObject(row.objectRef);
+      if (theObject is Instance) {
+        instance = theObject;
+      } else if (theObject is Sentinel) {
+        instance = null;
+        // TODO(terry): Tracking Sentinel's to be removed.
```

Comment:
is this a problem or can we just explain that the snapshot is a bit out of date as the object has been GCed?

Reply:
Not a problem but the object could have been GC'd since the snapshot.